engine/src/main/java/org/terasology/reflection/internal/TypeRegistryImpl.java

```diff
@@ -42,12 +51,36 @@
     public TypeRegistryImpl() {}
 
     public TypeRegistryImpl(ClassLoader classLoader) {
+        this();
         initializeReflections(classLoader);
     }
 
+    public TypeRegistryImpl(ModuleEnvironment environment) {
+        this();
+        reload(environment);
+    }
+
+    private static boolean filterWhitelistedTypes(String typeName) {
+        if (typeName == null) {
+            return false;
+        }
+
+        typeName = typeName.replace(".class", "");
+
+        int i = typeName.lastIndexOf('.');
+        if (i == -1) {
+            return false;
+        }
+
+        String packageName = typeName.substring(0, i);
+
+        return ExternalApiWhitelist.PACKAGES.contains(packageName) || WHITELISTED_CLASSES.contains(typeName);
+    }
+
     public void reload(ModuleEnvironment environment) {
```

Comment:
maybe move this into the interface so you can mock it for testing.

Reply:
That's a good idea. I'm thinking I might just get rid of the interface and make it a single class, since it isn't meant to be accessed outside the engine.